pkg/controllers/v1alpha1/datamigrate/implement.go

```diff
@@ -18,355 +18,133 @@ package datamigrate
-func (r *DataMigrateReconcilerImplement) reconcileCompleteDataMigrate(ctx cruntime.ReconcileRequestContext, targetDataMigrate datav1alpha1.DataMigrate) (ctrl.Result, error) {
-	log := ctx.Log.WithName("reconcileCompleteDataMigrate")
-	// 1. release lock on target dataset
-	err := r.releaseLockOnTargetDataset(ctx, targetDataMigrate, ctx.Dataset)
-	if err != nil {
-		log.Error(err, "can't release lock on target dataset", "targetDataset", ctx.Dataset)
-		return utils.RequeueIfError(err)
-	}
-
-	// 2. record event and no requeue
-	log.Info("DataMigrate complete, no need to requeue")
-	jobName := utils.GetDataMigrateJobName(utils.GetDataMigrateReleaseName(targetDataMigrate.Name))
-	r.Recorder.Eventf(&targetDataMigrate, v1.EventTypeNormal, common.DataMigrateJobComplete, "DataMigrate job %s succeeded", jobName)
-	return utils.NoRequeue()
+func (r *DataMigrateReconciler) SetTargetDatasetStatusInProgress(dataset *datav1alpha1.Dataset) {
+	dataset.Status.Phase = datav1alpha1.DataMigrating
 }
 
-// reconcileFailedDataMigrate reconciles DataMigrates that are in `DataMigratePhaseComplete` phase
-func (r *DataMigrateReconcilerImplement) reconcileFailedDataMigrate(ctx cruntime.ReconcileRequestContext, targetDataMigrate datav1alpha1.DataMigrate) (ctrl.Result, error) {
-	log := ctx.Log.WithName("reconcileFailedDataMigrate")
-	// 1. release lock on target dataset
-	err := r.releaseLockOnTargetDataset(ctx, targetDataMigrate, ctx.Dataset)
-	if err != nil {
-		log.Error(err, "can't release lock on target dataset", "targetDataset", ctx.Dataset)
-		return utils.RequeueIfError(err)
-	}
-
-	// 2. record event and no requeue
-	log.Info("DataMigrate failed, won't requeue")
-	jobName := utils.GetDataMigrateJobName(utils.GetDataMigrateReleaseName(targetDataMigrate.Name))
-	r.Recorder.Eventf(&targetDataMigrate, v1.EventTypeNormal, common.DataMigrateJobFailed, "DataMigrate job %s failed", jobName)
-	return utils.NoRequeue()
-}
-
-// releaseLockOnTargetDataset releases lock on target dataset if the lock currently belongs to reconciling DataMigrate.
-// We use a key-value pair on the target dataset's status as the lock. To release the lock, we can simply set the value to empty.
-func (r *DataMigrateReconcilerImplement) releaseLockOnTargetDataset(ctx cruntime.ReconcileRequestContext, targetDataMigrate datav1alpha1.DataMigrate, targetDataSet *datav1alpha1.Dataset) error {
-	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		currentRef := targetDataSet.GetDataOperationInProgress(cdatamigrate.DataMigrateLockName)
-		if currentRef != utils.GetDataMigrateRef(targetDataMigrate.Name, targetDataMigrate.Namespace) {
-			r.Log.Info("Found DataMigrateRef inconsistent with the reconciling DataMigrate, won't release this lock, ignore it", "targetDataSet", targetDataSet.Name)
-			return nil
-		}
-		datasetToUpdate := targetDataSet.DeepCopy()
-		datasetToUpdate.RemoveDataOperationInProgress(cdatamigrate.DataMigrateLockName)
```

Comment:
Another minor question here: `datasetToUpdate.RemoveDataOperationInProgress`'s implementation involves setting `dataset.Status.OperationRef[operationType] = ""`.

Why not using `delete(dataset.Status.OperationRef, operationType)`, I think it's more clear.

Reply:
fixed.